internal/translator/data/rules.go

```diff
@@ -34,6 +36,106 @@ type Rules struct {
 // RulesHelper has useful functions to work with the collection
 type RulesHelper interface {
 	FindByPattern(ctx context.Context, source, target, pattern string) ([]Rule, error)
+	NewRule(ctx context.Context, ruleStruct Rule) (*primitive.ObjectID, error)
+	FindRuleByPattern(ctx context.Context, sourceLanguage, targetLanguage, pattern string) ([]Rule, error)
+	UpdateOne(ctx context.Context, filter Rule, updateValue Rule) error
+	DeleteOne(ctx context.Context, rule Rule) error
+	DeleteMany(ctx context.Context, sourceLanguage, targetLanguage, pattern string) error
```

Comment:
Instead of pass three parameters would be worth to have a `filter` map like when you update.

Reply:
Is it easier to construct an object in the UI (filter) than only pass three strings?
Say me why?
The suggestion was taken in the code.

Reply:
If you are thinking in UI when you are writing this kind of things, maybe you are going in the wrong way.

High arity makes hardest the code to understand. If you have more than 2 params maybe you can join it in a Class or Struct in this case with Go.